server/routes/api/checkout.js

```diff
@@ -41,27 +41,54 @@ router.post('/create-transaction', async (req, res) => {
 // 3. Once user completes payment, will redirect back to `success_url` with
 //  a Stripe session_id included in the URL.
 
-router.post('/create-checkout-session', async (req, res) => {
-  try {
-    const acceptableCharges = [1, 2, 20, 50]
-    const { donationAmount } = req.body || {}
-    const parsedDonationAmount = parseInt(donationAmount, 10)
+router.post('/create-checkout-session', async (req, res) => {  
+  const { donationAmount } = req.body || {}
+  const origin = req.get('origin')
 
-    let donation
+  const input = format(donationAmount);
+  const [inputIsValid, message] = validate(input);
+  const donationAmountForStripe = input * 100; // Stripe accepts values in cents
 
-    if (parsedDonationAmount < 2) {
-      // TODO: Change to something better later.
-      donation = 150
-    } else {
-      donation = parsedDonationAmount * 100
-    }
+  if (inputIsValid) {
+    const session = checkout(donationAmountForStripe, origin);
+    // console.log('session:', session)
 
-    if (!acceptableCharges.includes(parsedDonationAmount)) {
-      return res.status(400).send({ error: 'Invalid Amount' })
+    if (session.error) {
+      console.log({ error: session.error, message: 'An error occurred with Stripe checkout' });
+      return res.json(session);
     }
 
-    const origin = req.get('origin')
+    return res.json({ url: session.url, sessionId: session.id })
+  } else {
+    return res.status(400).send({ error: 'Invalid Amount; did not create Stripe checkout session', message })
+  }
+})
+
+// format input value
+function format(value) {
+  // separating parameter assignment and parseFloat operation for consistent outcome on change
+  value = parseFloat(value); // outputs: number
+  value = value.toFixed(2); // outputs: string
+  value = parseFloat(value); // outputs: number
+  return value;
+}
+
+// validate input value
+function validate(value = this.customDonationAmount) {
+  let inputIsValid = true, message = "";
 
+  if (value > 1.49 && value < 10000.01) return [inputIsValid, message];
+  
+  if (isNaN(value)) message = "Please select or enter a valid amount.";
+  if (value < 1.5) message = "Please enter a donation amount higher than $1.50.";
+  if (value > 10000) message = "Amplify currently only accept donation amounts less than $10,000."
+
+  inputIsValid = false;
+  return [inputIsValid, message]; // i.e. [ false, "Please select or enter a valid amount." ]
+}
+
+async function checkout(donationAmount, origin) {
```

Comment:
I don't think you need to separate the checkout logic from the route method. In this case, the route method's purpose is to create the checkout.

Reply:
Noted. I can incorporate this back to be directly within the post req without breaking it out into a function.